### PR TITLE
Bugfix/#2987

### LINKED ISSUE #2987: Product on Wishlist does not link to its variant when using URL Dispatcher
## Current behavior
With URL Dispatcher on, when saving a specific product variant to the wishlist, link is not generated correctly (e.g. with ?ChildSku).

## Expected behavior
As before without URL Dispatcher - the wishlist product should link to its correct variant.

## Steps to reproduce the issue
1. Add non-default product-variation to wishlist
2. Open wishlist and click product, which only links to the Configurable product without ChildSku

## Can you handle fixing this bug by yourself?

Maybe, but not right now.

## Which [Release Cycle](https://docs.vuestorefront.io/guide/basics/release-cycle.html) state this refers to? Info for developer.
Pick one option.

- [x] This is a bug report for test version on https://test.storefrontcloud.io - In this case Developer should create branch from `develop` branch and create Pull Request `2. Feature / Improvement` back to `develop`.
- [x] This is a bug report for current Release Candidate version on https://next.storefrontcloud.io - In this case Developer should create branch from `release` branch and create Pull Request `3. Stabilisation fix` back to `release`.
- [x] This is a bug report for current Stable version on https://demo.storefrontcloud.io and should be placed in next stable version hotfix - In this case Developer should create branch from `hotfix` or `master` branch and create Pull Request `4. Hotfix` back to `hotfix`.

### PR DESCRIPTION
**Pull request by michasik**
### Related issues
#2987 

closes #2987 

### Short description and why it's useful
1) Now the url path for the product from **Wishlist** and **MicroCart** components is generated by _formatProductLink_ method from _"core\modules\url\helpers\index.ts"_ (as it has already been implemented in **Compare** component).

2) The _formatProductLink_ was changed, now it checks if the product has populated _options_ property, not _configurable_children_ property.
The reason for that change is that products from cart didn't have _configurable-children_ property.
It also makes shorter url path for non-configurable products by not adding any query params. For example for the **roller** the path is:
_".../gear/gear-3/sprite-foam-roller-22.html"_
not
_"../gear/gear-3/sprite-foam-roller-22.html?childSku=24-WG088"_
which is shorter and I guess more SEO friendly.

### Which environment this relates to
Check your case. In case of any doubts please read about [Release Cycle](https://docs.vuestorefront.io/guide/basics/release-cycle.html)

- [x] Test version (https://test.storefrontcloud.io) - this is a new feature or improvement for Vue Storefront. I've created branch from `develop` branch and want to merge it back to `develop`
- [ ] RC version (https://next.storefrontcloud.io) - this is a stabilisation fix for Release Candidate of Vue Storefront. I've created branch from `release` branch and want to merge it back to `release`
- [ ] Stable version (https://demo.storefrontcloud.io) - this is an important fix for current stable version. I've created branch from `hotfix` or `master` branch and want to merge it back to `hotfix`

### Upgrade Notes and Changelog

- [ ] No upgrade steps required (100% backward compatibility and no breaking changes)
- [x] I've updated the [Upgrade notes](https://github.com/DivanteLtd/vue-storefront/blob/develop/doc/Upgrade%20notes.md) and [Changelog](https://github.com/DivanteLtd/vue-storefront/blob/develop/CHANGELOG.md) on how to port existing VS sites with this new feature

**IMPORTANT NOTICE** - Remember to update `CHANGELOG.md` with description of your change

### Contribution and currently important rules acceptance

- [x] I read and followed [contribution rules](https://github.com/DivanteLtd/vue-storefront/blob/master/CONTRIBUTING.md)
- [x] I read the [TypeScript Action Plan](https://github.com/DivanteLtd/vue-storefront/blob/master/doc/TypeScript%20Action%20Plan.md) and adjusted my PR according to it
- [x] I read about [Vue Storefront Modules](https://github.com/DivanteLtd/vue-storefront/blob/master/doc/api-modules/about-modules.md) and I am aware that every new feature should be a module

### CODE REVIEW

File: src/themes/default/components/core/blocks/Wishlist/Product.vue [hunk 1]
```diff
@@ -3,7 +3,6 @@
     <div @click="closeWishlist">
       <router-link :to="localizedRoute({
         name: product.type_id + '-product',
-        fullPath: product.url_path,
```
Comment (pkarw) [inline src/themes/default/components/core/blocks/Wishlist/Product.vue]: I don't think this is the right solution. By setting `fullPath` we inform `localizedRoute` to use the `localizedDispatcherRoute` that generates the seo-friendly URL. It should consist the `?childSku` parameter anyway. It's done by this line: https://github.com/DivanteLtd/vue-storefront/blob/20b693e28eeac810252593e1b42e709cc03701e9/core/lib/multistore.ts#L137

If it's not there (the childSku) we need to find the reason

**Comment by pkarw** [comment]
The error is with this line: https://github.com/DivanteLtd/vue-storefront/blob/20b693e28eeac810252593e1b42e709cc03701e9/core/lib/multistore.ts#L116 - where params are cleared-out. It's to prevent adding the other parameters (like `sku`) to the query string.

The more "shallow" solution to this issue is to change the way we call `router-link` in the Whislist.vue (using `query-string` and adding the `childSku` at this level)

**Comment by michasik** [comment]
> We need to keep the `fullPath`

_fullPath_ is preserved in url helpers now

**Comment by pkarw** [comment]
This is a pretty cool refactor. Please just check one thing: the `product.configurable_children` is not set in the cart - and using `product.options` in that case is clever trick :) However, on the product list we don't have `options` - and only the `configurable_children`. If user is using filters on category page - for example filtering the Red blouses - we were using these `configuable_children` check to add the `?childSku=product.sku` which for example is `MH06-XS-Yellow`. 

Not sure if by this change you didn't remove this feature? :) Can you please doublecheck?

**Comment by pkarw** [comment]
oh, and please do resolve the conflicts - then: it's ready to be merged in :)

**Comment by michasik** [comment]
@pkarw you were right, from filtered products list the `formatProductLink` method was returning wrong url path. Now I have added back the `configurable_children` property to this method alongside the `options` property.
I tested navigation from category, wish list, cart and compare view - works fine now.
Conflicts resolved :-)

**Comment by pkarw** [comment]
Thank you!

**Comment by alinadivante** [comment]
@michasik there is a problem with URL path for product from the wishlist (I'm not redirected to the right configurable product)
And something else - when I add products to the wishlist / compare lsit and go to /de store - then when trying to open a product from the list, you will be redirected to https://next.storefrontcloud.io/page-not-found

**Comment by michasik** [comment]
@alinadivante 
1. I tested locally on develop branch and it works correctly. Tested scenarios:
 - From wishlist / cart / compare / filtered category to configured product
2. I cannot test this scenario locally because I have 0 products with `de` storecode... 😞

**Comment by alinadivante** [comment]
@michasik sorry for that. Your PR has milestone 1.11.0-rc.1 ! And I was focused on 1.10.0-rc.1 ;) my mistake

**Pull request merged**